tool/tsh/db.go

```diff
@@ -813,6 +807,13 @@ func getDatabase(cf *CLIConf, tc *client.TeleportClient, dbName string) (types.D
 }
 
 func needDatabaseRelogin(cf *CLIConf, tc *client.TeleportClient, database *tlsca.RouteToDatabase, profile *client.ProfileStatus) (bool, error) {
+	if cf.LocalProxyTunnel {
+		// Don't login to database here if local proxy tunnel is enabled.
+		// When local proxy tunnel is enabled, the local proxy will check if DB login is needed when
+		// it starts and on each new connection.
+		return false, nil
+	}
+
```

Comment:
Does the local proxy OnStart check if re-login is needed because of different `--db-user` or `--db-name`?

Reply:
I don't think different --db-user or --db-name will be possible, because the local proxy tunnel will have certs in memory each time. I suppose the `--cert-file` and `--key-file` could be passed though, so in that case I can add cert subject checking

Reply:
ah, ok. so the local proxy is configured with zero certs and `OnStart` will always fetch a new one? So there is some performance penalty that we are refetching certs even though it might not be necessary (e.g. non MFA flows). I don't think it's a big deal, but if it's something easy we can do to optimize the performance that would be great.

Reply:
Prior behavior would attempt to load certs from disk if they were already there - I don't see anything wrong with adding that back. You're right that it's an extra cert fetch in a non-MFA flow currently. But I will keep this change for `needDatabaseRelogin` so proxy tunnels don't trigger a "normal" login, because it's simpler logic to keep track of.